src/entry-editable.ts

```diff
@@ -8,27 +8,91 @@ export function addTags(entry: EntryModel, contentTypeUid: string, tagsAsObject:
 function getTag(content: object, prefix: string, tagsAsObject: boolean, locale: string): object {
     const tags: any = {}
     Object.entries(content).forEach(([key, value]) => {      
+        if (key === '$') return
+
         switch (typeof value) {
             case "object":
                 if (Array.isArray(value)) {
                     value.forEach((obj, index) => {
+                        const childKey = `${key}__${index}`
+                        const parentKey = `${key}__parent`
+                        /**
+                         * Indexes of array are handled here
+                         * {
+                         *  "array": ["hello", "world"],
+                         *  "$": {
+                         *      "array": {"data-cslp": "content_type_uid.entry_uid.locale.array"}
+                         *      "array__0": {"data-cslp": "content_type_uid.entry_uid.locale.array.0"}
+                         *      "array__1": {"data-cslp": "content_type_uid.entry_uid.locale.array.1"}
+                         *  }
+                         * }
+                         */
+                        tags[childKey] = getTagsValue(`${prefix}.${key}.${index}`, tagsAsObject)
+                        tags[parentKey] = getParentTagsValue(`${prefix}.${key}`, tagsAsObject)
                         if (typeof obj !== 'undefined' && obj !== null && obj._content_type_uid !== undefined && obj.uid !== undefined) {
+                            /**
+                             * References are handled here
+                             * {
+                             *  "reference": [{
+                             *      "title": "title",
+                             *      "uid": "ref_uid",
+                             *      "_content_type_uid": "ref_content_type_uid",
+                             *     "$": {"title": {"data-cslp": "ref_content_type_uid.ref_uid.locale.title"}}
+                             *  }]
+                             * }
+                             */
                             value[index].$ = getTag(obj, `${obj._content_type_uid}.${obj.uid}.${obj.locale || locale}`, tagsAsObject, locale)
-                        }else {
-                            if (typeof obj === "object") {
-                                obj.$ = getTag(obj, `${prefix}.${key}.${index}`, tagsAsObject, locale)
-                            } else {
-                                tags[key] = getTagsValue(`${prefix}.${key}`, tagsAsObject)
-                            }
+                        }else if (typeof obj === "object") {
+                            /**
```

Comment:
here @abhinav-from-contentstack

Reply:
keeping the comments as they help understanding code function